### PR TITLE
Added .editorconfig

### LINKED ISSUE #199: Add .editorconfig file
The `.editorconfig` file would help maintain consistency of code for the project.

### PR DESCRIPTION
**Pull request by aashutoshrathi**
Fixes #199 
Closes #204 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [ ] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

**Comment by aashutoshrathi** [comment]
@aviaryan @singhpratyush take a look

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+[{*.py, *.go}]
+charset = utf-8
+indent_style = tab
```
Comment (aviaryan) [inline .editorconfig]: pep8 follows space indentation for Python.

Comment (aashutoshrathi) [inline .editorconfig]: so should I make it space and indet_size = 4 ?

**Comment by aashutoshrathi** [comment]
for C/C++ and JAVA, 
should it be same as GO ?

**Comment by singhpratyush** [comment]
Java: 4 space indentation
C/C++: tab indentation - length 4

Other properties as usual.

**Comment by aashutoshrathi** [comment]
@singhpratyush updated, take a look

**Pull request merged**